### PR TITLE
UGENE-5000. Incorrect behavior of "Sequence from file" action

### PR DESCRIPTION
**Pull request by yalgaer**

### CODE REVIEW

File: src/corelibs/U2Core/src/tasks/AddSequencesToAlignmentTask.cpp [hunk 1]
```diff
@@ -186,38 +186,47 @@ void AddSequencesFromFilesToAlignmentTask::prepare() {
...
 }
 
 QList<Task *> AddSequencesFromFilesToAlignmentTask::onSubTaskFinished(Task *subTask) {
-    const QList<Task *> emptySubTasks;  // Helper constant. This method never returns any subtasks.
-
     propagateSubtaskError();
-    if (isCanceled() || hasError()) {
-        return emptySubTasks;
-    }
+    CHECK(!isCanceled() && !hasError(), {});
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2Core/src/tasks/AddSequencesToAlignmentTask.cpp]: Could be even better - `CHECK_OP(stateInfo, {});`

File: src/corelibs/U2Core/src/tasks/AddSequencesToAlignmentTask.cpp [hunk 2]
```diff
@@ -186,38 +186,47 @@ void AddSequencesFromFilesToAlignmentTask::prepare() {
...
+
+    auto *loadDocumentSubTask = qobject_cast<LoadDocumentTask *>(subTask);
```
Comment (DmitriiSukhomlinov) [inline src/corelibs/U2Core/src/tasks/AddSequencesToAlignmentTask.cpp]: `auto` could be a pointer itself, no need of `auto *`

**Review by DmitriiSukhomlinov — DISMISSED** [review]

File: src/corelibs/U2Core/src/tasks/AddSequencesToAlignmentTask.cpp (hunk 1 above)
Comment (yalgaer) [reply src/corelibs/U2Core/src/tasks/AddSequencesToAlignmentTask.cpp]: Fixed

File: src/corelibs/U2Core/src/tasks/AddSequencesToAlignmentTask.cpp (hunk 2 above)
Comment (yalgaer) [reply src/corelibs/U2Core/src/tasks/AddSequencesToAlignmentTask.cpp]: Right, missed that symbol

**Pull request merged**